arangod/RestServer/CheckVersionFeature.cpp

```diff
@@ -142,20 +142,26 @@ void CheckVersionFeature::checkVersion() {
   bool ignoreDatafileErrors = databaseFeature.ignoreDatafileErrors();
 
   // iterate over all databases
-  for (auto& name : databaseFeature.getDatabaseNames()) {
-    TRI_vocbase_t* vocbase = databaseFeature.lookupDatabase(name);
-    methods::VersionResult res = methods::Version::check(vocbase);
+  for (auto const& name : databaseFeature.getDatabaseNames()) {
+    auto vocbase = databaseFeature.useDatabase(name);
+    // during this phase, there should be no concurrent dropping of databases,
```

Comment:
Same question

Reply:
There should not be any concurrent deletions while this code is executed. The CheckVersionFeature only runs at server startup.
The previous version of the code called `lookupDatabase(...)` and then unconditionally dereferenced the return value, which was unsafe. The version here avoids any potential issues by checking the return value before deferencing them.